src/lib/proxy-server.ts

```diff
@@ -99,35 +99,21 @@ async function handleProxyRequest(
 }
 
 /**
- * On Windows, node doesn't throw an error if port is busy, so we use the net module to explicitly check
- * if it's possible to establish a TCP connection to that port (meaning it's busy).
+ * On Windows, node doesn't throw an error if port is busy, so we use portFinder to check
+ * if the port is available.
  */
 export async function checkPortInWindows( port: number ): Promise< boolean > {
 	if ( process.platform !== 'win32' ) {
 		return true;
 	}
```

Comment:
Not a requirement, but coming back to this code, I would argue that we should run this logic on all platforms (not just on Windows). Yes, `http.createServer` already throws an error on macOS, but making Studio's business logic consistent across platforms is a way to avoid future headaches. This PR seems like a good opportunity to address that.

Reply:
If we make this change, I'd also throw an error with the `PROXY_ERROR_PORT_IN_USE` directly from this function, and adjust the `catch` clause in `startProxyServer` to rethrow the error.

Reply:
Excellent suggestions @fredrikekelund! I will implement those changes.

Reply:
Addressed your feedback in [b1f2b46](https://github.com/Automattic/studio/pull/1281/commits/b1f2b46de512fc4ad21be85a403620a298ae2ca9)!